internal/receiver/smartagentreceiver/receiver_test.go

```diff
@@ -242,3 +264,102 @@ func TestConfirmStartingReceiverWithInvalidMonitorInstancesDoesntPanic(t *testin
 		})
 	}
 }
+
+func TestSmartAgentConfigProviderOverrides(t *testing.T) {
+	t.Cleanup(cleanUp)
+	cfg := newConfig("valid", "cpu", 1)
+	observedLogger, logs := observer.New(zapcore.WarnLevel)
+	logger := zap.New(observedLogger)
+	r := NewReceiver(logger, cfg, consumertest.NewMetricsNop())
+	host := &mockHost{
+		smartagentextensionConfig:  getSmartAgentExtensionConfig(t),
+		smartagentextensionConfig2: getSmartAgentExtensionConfig(t),
```

Comment:
Would it make more sense to have two extension definitions in the config file with different values and verify we're using the one we expect?